fission-core/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![warn(missing_debug_implementations, rust_2018_idioms)]
+#![deny(unreachable_pub, private_in_public)]
```

Comment:
@zeeshanlakhani Not sure if we need all of these for `core`, thoughts?

Reply:
I mean, core will still be a crate, yeah? These are more forcing functions for devs to use docs, check for unreachability, etc. Obviously tunable, but helpful for consistency.